src/Riok.Mapperly/Descriptors/MappingBuilders/DictionaryMappingBuilder.cs

```diff
@@ -66,6 +77,11 @@ public static class DictionaryMappingBuilder
         if (BuildKeyValueMapping(ctx) is not var (keyMapping, valueMapping))
             return null;
 
+        // if target is an immutable dictionary then don't a foreach loop
+        if (SymbolEqualityComparer.Default.Equals(ctx.Target.OriginalDefinition, ctx.Types.ImmutableDictionaryT)
+            || SymbolEqualityComparer.Default.Equals(ctx.Target.OriginalDefinition, ctx.Types.ImmutableSortedDictionaryT))
```

Comment:
Instead of checking the immutable types, couldn't we check if the `IImmutableDictionary<TKey, TValue>` is implemented? 🤔 
Also I think we should use the same approach as for the enumerables. Eg. if we do the denylist here, we should adjust the enumerables too.

Reply:
Good point I'll make the change.

 I'll add the denylist in #341